### DISCUSSION TITLE
InputPicker icon

### DISCUSSION THREAD

**Discussion by PiousVenom** [rsuite/rsuite#2077] [Q&A]
Is it possible to change the icon on the InputPicker? I'd like to use something other than the generic down caret.

**Comment by SevenOutman** [comment]
You can pass a component to `caretComponent` prop

```jsx
// MyCaret have to accept `className` prop so that RSuite can apply styles on it
const MyCaret = (props) => <span {...props}>⬇️</span>;

<InputPicker
  caretComponent={MyCaret}
/>
``` 

This is missing from the documentation, we need to fix it.

**Reply by PiousVenom** [reply]
Whenever I do this, I get a "property caretComponent does not exist" error. I am using RSuite 5.1.0.

**Reply by SevenOutman** [reply]
Looks like we need to fix the type definition.

As a workaround, you could pass the un-defined props by spreading an object that casts to `any` (suppose you're using typescript).

```jsx
// MyCaret have to accept `className` prop so that RSuite can apply styles on it
const MyCaret = (props) => <span {...props}>⬇️</span>;

const props = {
  caretComponent: MyCaret
};

<InputPicker
  {...(props as any)}
/>
```

**Reply by PiousVenom** [reply]
Is it not possible to just pass in an SVG element to be used instead? I'm really new to React and TypeScript, so this is just a bit beyond my understanding for the moment.

**Reply by SevenOutman** [reply]
No it's not possible with the API we currently provide. But we may improve the API design if we find a solution that's easier to use without compromising flexibility in future.

**Reply by PiousVenom** [reply]
Okay, thank you. I'll see what I can figure out then.
